modes/settings_menues.h

```diff
@@ -106,7 +106,7 @@ struct ChangeBladeLengthBlade1 : public SPEC::MenuBase {
     SPEC::MenuBase::exit();
   }
 
-  int getLength() { return this->pos_; }
+  int getLength() { return this->pos_ + 1; }
```

Comment:
I think it would be better to have getLength return a zero-indexed value and fix ShowLengthStyle instead.

Reply:
Actually, I don't think this is correct.
ShowLengthStyle is supposed to light up the last LED in the blade.
If this returns pos_ + 1, then a zero-length blade would light up the first led, which is incorrect.
A zero-length blade should not show anything with ShowLengthStyle.

We could change the menu to not allow the value zero, but then we need to change everywhere that pos_ is set or read.

Reply:
I thought I had sorted this and was going to update but now I have to ask a few questions first.
Are you saying that the range should allow for the blade to be set to zero?.

I was going to make it so that if the maxlen is set as zero, we can just bail and return to the previous menu.
That's easy, but if the maxlen is not zero, and we want to allow the blade length to be set to a zero value, that makes it harder because now size() would need to account for an additional position?

To confirm(?) blade length is 1-based indexing. pos_ is zero-based.
So I think if we just set this->pos_ to the blade length in mode_activate, all the +1 stuff after doesn't need to be.

Where you said "ShowLengthStyle is supposed to light up the last LED in the blade." Do you mean upon entering the mode, the maxlen should be where we start?
I was thinking we set it so that the saved length is where we start. 

Also, when would there ever be a zero-length blade? I guess whether or not that would make sense or not doesn't matter, we just need to accommodate the possibility?

Reply:
Well I figured out (one way) how to do this, including accomodating a zero length blade, but it required overriding next() and prev() to the point I'm not sure there's much reason to inherit from MenuBase anymore.

Replying to my own questions
> Where you said "ShowLengthStyle is supposed to light up the last LED in the blade." Do you mean upon entering the mode, the maxlen should be where we start?

Answered for myself based on the code and how it's working. This meant the last LED of the current (or saved) blade length, if none saved, it uses the maxlen.
 

> Are you saying that the range should allow for the blade to be set to zero?.
Well, it can if you want. See latest update to this PR.



> To confirm(?) blade length is 1-based indexing. pos_ is zero-based.

Yes. So to include zero as a valid blade length option, we can omit the modolo and manually calculate when to wrap instead by checking if the next() or prev() operation would result in overshooting the desired range values, and if so, setting the pos_ directly to those values.
If the user has manually set the blade length to 0 in the config, just exit as no length adjustment is possible.


> Also, when would there ever be a zero-length blade?

Maybe if there were a blade detect setup where the user decided to make a zero length for the non-inserted blade state?

Reply:
> I thought I had sorted this and was going to update but now I have to ask a few questions first.
Are you saying that the range should allow for the blade to be set to zero?.

No. I'm saying that my original assumption was that there was a 0-index vs. 1-index issue going on here, which is subtly different from allowing a zero length. My original assumption was wrong.

> I was going to make it so that if the maxlen is set as zero, we can just bail and return to the previous menu.
That's easy, but if the maxlen is not zero, and we want to allow the blade length to be set to a zero value, that makes it harder because now size() would need to account for an additional position?

maxlen should never be zero. That would be a string with zero LEDs.
And as I said, I did not intend that a length of zero should be allowed.

> To confirm(?) blade length is 1-based indexing. pos_ is zero-based.

No. blade length is not really 1-based, but if we don't want to allow the zero value, then we still need an offset between pos_ and length.  (This might seem like a minor point, but it matters when we calculate size())

> Where you said "ShowLengthStyle is supposed to light up the last LED in the blade." Do you mean upon entering the mode, the maxlen should be where we start?

No, I mean that if length is 1, then the first LED in the string should be lit up.
If length is 2, then the second LED of the string should light up.